components/raftstore-v2/src/operation/command/write/ingest.rs

```diff
@@ -67,12 +67,32 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         ctx: &mut StoreContext<EK, ER, T>,
         ssts: Box<[SstMeta]>,
     ) {
-        let epoch = self.region().get_region_epoch();
         let mut stale_ssts = Vec::from(ssts);
-        stale_ssts.retain(|sst| util::is_epoch_stale(sst.get_region_epoch(), epoch));
+        let skip = (|| {
+            fail::fail_point!("on_cleanup_import_sst", |_| false);
+            true
+        })();
+        if skip {
+            let epoch = self.region().get_region_epoch();
+            stale_ssts.retain(|sst| util::is_epoch_stale(sst.get_region_epoch(), epoch));
+        }
+
+        // some sst needs to be kept if the log didn't flush the dish.
+        if let Some(log_recovery) = self.storage().apply_trace().log_recovery() {
```

Comment:
apply_trace.data_cfs may be sufficient here? As for a specified cf, apply will be skipped if apply index is less than flushed index.

Reply:
that sst will not record in the sst apply state.